core/src/main/scala/cats/Parallel.scala

```diff
@@ -312,6 +312,18 @@ object Parallel extends ParallelArityFunctions2 {
     P.sequential(ftab)
   }
 
+  /**
+   * Like `Foldable[A].foldMapA`, but uses the applicative instance
+   * corresponding to the Parallel instance instead
+   */
+  def parFoldMapA[T[_]: Foldable, M[_], A, B: Monoid](
+    ta: T[A]
+  )(f: A => M[B])(implicit P: Parallel[M]): M[B] = {
```

Comment:
I think we've made a mistake in allowing context bounds to be mixed with explicit implicit parameter sections. In a case like this with multiple context bounds it's especially confusing.

In general it tends to be best for inference for implicit parameters should be in the same order as their types appear in the arguments, which in this case would put the `Parallel` before the `Monoid`, but as it's currently written they're switched. I think the following would be better:

```scala
  def parFoldMapA[T[_], M[_], A, B](
    ta: T[A]
  )(f: A => M[B])(implicit T: Foldable[T], P: Parallel[M], B: Monoid): M[B] = {
```

Reply:
Done (and I did the same in the syntax wrapper).